src/main/java/com/google/sps/servlets/ProfileProjectsServlet.java

```diff
@@ -0,0 +1,76 @@
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.SearchProject;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.*;
+import com.google.gson.Gson;
+
+// Author: David V.P.
+@WebServlet("/profile-projects")
+public class ProfileProjectsServlet extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        String email = request.getParameter("user_id");
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Query query = new Query("Project").addSort("timestamp", SortDirection.DESCENDING);
+        Iterable<Entity> results = datastore.prepare(query).asIterable();
+
+        List <SearchProject> projects = new ArrayList<>();
+        for (Entity entity : results) {
+            String author = (String) entity.getProperty("author");
+
+            // only add project if author matches the email
+            if(email.equals(author)){
+
+                //Get the properties of the entity:
```

Comment:
Mannie changed the project property names. Remember to update them. You can see the new ones implemented in my latest Pull request.

Reply:
updated them, thank you.

Reply:
The best way to do this (you odn't have to do it here if you don't want to) is to have a constants file that defines all these strings there. This way there's no way to have discrepancies in the names used in different files.